### ISSUE TITLE
Use example for initializing pre-trained network with weights parameter

### ISSUE THREAD

**Issue by etch32** [aigamedev/scikit-neuralnetwork#188]
From the docs:
weights: list of tuple of array-like, optional

A list of (weights, biases) tuples to be reloaded for each layer, in the same order as layers was specified. Useful for initializing with pre-trained networks.

I trained a network, e.g.

```python
model = Classifier(
    layers=[
        Layer("Sigmoid", units=4),
        Layer("Softmax")],
    learning_rate=0.01,
    n_iter=25,
    random_state=1,
    verbose=False)

model.fit(X_train, y_train)

# save the weights

ps = model.get_parameters()
weights = [(ps[0].weights, ps[0].biases), (ps[1].weights, ps[1].biases)]
```
Want to reinitialize model with these weights (e.g. in a new session)
```python
model2 = Classifier(
    layers=[
        Layer("Sigmoid", units=4),
        Layer("Softmax")],
    learning_rate=0.01,
    n_iter=25,
    random_state=1,
    weights=weights,    # initialize with previously trained weights
    verbose=False)
```
 Then felt I should just be able to call predict
```python
y_pred = model2.predict(X_test)
```
But get the following error.
   "Can't predict without fitting: output classes are unknown."
AssertionError: Can't predict without fitting: output classes are unknown.

OK, the part about the output classes are unknown seems reasonable. However if I call fit() then it will just re-compute the weights and use these new weights.

I may be misinterpreting the intent/use of the weights parameter.

**Comment by alexjc** [comment]
If you're just serializing you should probably use `pickle`, it's tested :-) I think I only used weights with `Regressor` classes, so it's a bug.

I think you can also call `_initialize(X, y)` to initialize the classifier correctly. If it works, then we probably need a way to pass in the classes to the constructor too, or just disable weights for `Classifier`?

**Comment by etch32** [comment]
Thanks for the quick reply Alex. I often do pickle a model and it works well. In this case I'm not wanting to have a disk dependancy so wanted to just store the weights in my predict script.

So I tried your suggestion. After some wrangling I got the _initialize() method to work (even though my y is a single column of binary 1 or 0, _initialize was expecting it to be one-hot encoded--I think). However, even after _initialize() I get the same AssertionError as above.

**Comment by alexjc** [comment]
Can you paste the traceback? I don't have time to reproduce it, but maybe that'll help...

**Comment by etch32** [comment]
Here you go. Call to _initialize with one-hot encoded y:

    model2._initialize(X_test, yoh_test)

Then the call to predict which generates the stack trace

```python
 File "train_models.py", line 207, in run_nn
    y_pred = model2.predict(X_test)
  File ".../lib/python2.7/site-packages/sknn/mlp.py", line 459, in predict
    "Can't predict without fitting: output classes are unknown."
AssertionError: Can't predict without fitting: output classes are unknown.
```

Thanks for looking into it.
Eric

**Comment by shangwang99** [comment]
Hi @alexjc I encountered a similar problem, after using clf.set_parameters( [(ps[0].weights, ps[0].biases), (ps[1].weights, ps[1].biases)]) to set parameters of a classifier in a new session, I got the following error, "Can't predict without fitting: output classes are unknown."
AssertionError: Can't predict without fitting: output classes are unknown.

And I checked clf.is_initialized, it returned False. May I know how to deal with this problem? Or could anyone provide an example on using set_parameters() to initialize a classifier to make prediction?

Thank you very much. ^^
